request.js

```diff
@@ -1428,15 +1428,15 @@ Request.prototype.json = function (val) {
     if (self.body !== undefined) {
       if (!/^application\/x-www-form-urlencoded\b/.test(self.getHeader('content-type'))) {
         self.body = safeStringify(self.body)
+      } else {
+        self.body = rfc3986(self.body)
```

Comment:
Can you swap this conditional around?  Also, I see we have a test for it (`body + header + json`) but in what situation  is this needed?